src/Features/CustomerEffortScoreTracks.php

```diff
@@ -90,15 +112,28 @@ private function get_product_count() {
 	 * Enqueue the CES survey trigger for a new product.
 	 */
 	private function enqueue_ces_survey_for_new_product() {
+		// Only add the JS to trigger the CES modal if
+		// this modal hasn't been dismissed or actioned yet for this action.
+		$shown_for_features = get_option( self::SHOWN_FOR_ACTIONS_OPTION_NAME, array() );
```

Comment:
It looks like we should establish a pattern similar to the inbox notes for checking if a CES prompt has been shown before, to keep this DRYer.

Reply:
I agree, I tend to stick to the rule of three for code reuse, but it makes sense to refactor this now. Are you happy for me to do this in a follow-up PR?

Reply:
I've done this in fc8fffa